shellstub.py

```
@@ -0,0 +1,90 @@
+        if variants is None:
+            variants = [dict(match=None, stdout=stdout, stderr=stderr, exit_code=exit_code)]
+        for v in variants:
+            if callable(v.get("match")):
+                raise NotImplementedError("callable matchers not supported")
+        spec = {"variants": variants}
+        self._specs[name] = spec
+        path = self.dir / name
+        path.write_text(self._wrapper_source(name, spec))
+        path.chmod(0o755)
+
+    def calls_of(self, name: str) -> list[Call]:
```

Pylint (3.3.7)

[convention] Missing function or method docstring

(C0116)